### PR TITLE
Feat: Adds `PaymentMethods` component

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

- [x] Moves `PaymentMethods ` component to @faststore/components 
- [x] Adds `PaymentMethods ` styles on @faststore/ui
- [x] Refactors `PaymentMethods `
- [x] Adds `PaymentMethods ` doc page
- [x] Adds Methods Payments flags icons

## How to test it?

run yarn start in app/docs folder and checks the `PaymentMethods ` component or test it using the doc preview link:
- [PaymentMethods Component](https://faststore-site-git-feat-payment-methods-fs-767-faststore.vercel.app/components/organisms/payment-methods)

### Starter PR
[Feat: Replace PaymentMethods FSUI](https://github.com/vtex-sites/nextjs.store/pull/370)

### CODE REVIEW

File: apps/site/pages/components/organisms/payment-methods.mdx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+Import Styles
+
+```tsx
+import '@faststore/ui/src/components/organisms/PaymentMethods/styles.scss'
+```
+
+---
+
+export const flags = [
```
Comment (renatamottam) [inline apps/site/pages/components/organisms/payment-methods.mdx:54]: I think this should be shown on `Code` tab below.

File: packages/ui/src/components/organisms/PaymentMethods/styles.scss [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  // --------------------------------------------------------
+  // Structural Styles
+  // --------------------------------------------------------
+
+  width: 100%;
+
+  [data-fs-payment-methods-title] {
+    & > *:first-child {
+      margin-bottom: var(--fs-payment-methods-title-margin-bottom);
```
Comment (renatamottam) [inline packages/ui/src/components/organisms/PaymentMethods/styles.scss:29]: This `margin-bottom` doesn't seems to be needed

File: packages/ui/src/components/organisms/PaymentMethods/styles.scss [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    @include media("<notebook") {
+      text-align: center;
+    }
+  }
+
+  [data-fs-payment-methods-flags] {
+    display: grid;
+    grid-template-columns: repeat(4, minmax(0, max-content));
+    row-gap: var(--fs-spacing-1);
```
Comment (renatamottam) [inline packages/ui/src/components/organisms/PaymentMethods/styles.scss:43]: Suggestion: use `margin-top` and `row-gap` as tokens.

File: packages/components/src/assets/index.ts [hunk 4]
```diff
@@ -1,3 +1,6 @@
+// Payment Flags Icons
+export * from './PaymentFlags'
```
Comment (renatamottam) [inline packages/components/src/assets/index.ts:2]: Fan, I think we should explain on the doc how the dev could add a new icon as PaymentFlags

**Review by renatamottam — COMMENTED** [review]
Nice work! Left a few suggestions :)

File: packages/components/src/assets/index.ts (hunk 4 above)
Comment (hellofanny) [reply packages/components/src/assets/index.ts:2]: oh yeah! Lemme add.

**Review by renatamottam — APPROVED** [review]

File: apps/site/pages/components/organisms/payment-methods.mdx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  - UI library
+  - ecommerce
+  - react
+sidebar_custom_props:
+  image: https://vtexhelp.vtexassets.com/assets/docs/src/PaymentMethods___b0f344b254360daa53406cd631887adf.png
+---
+
+import { Tab, Tabs } from 'nextra-theme-docs'
+import PropsSection from 'site/components/PropsSection'
+import { TokenTable, TokenRow, TokenDivider } from 'site/components/Tokens'
```
Comment (eduardoformiga) [inline apps/site/pages/components/organisms/payment-methods.mdx:13]:
```suggestion
import { TokenTable, TokenRow } from 'site/components/Tokens'
```

**Review by eduardoformiga — APPROVED** [review]
Nice job!

**Pull request merged**